src/librustc_mir/const_eval/eval_queries.rs

```diff
@@ -226,7 +226,7 @@ pub fn const_eval_validated_provider<'tcx>(
 ) -> ::rustc::mir::interpret::ConstEvalResult<'tcx> {
     // see comment in const_eval_raw_provider for what we're doing here
     if key.param_env.reveal == Reveal::All {
-        let mut key = key.clone();
+        let mut key = key;
```

Comment:
And I guess these too? Though here at least we declare it as mutable.

Reply:
We need the `mut` copy here due to the `key.param_env.reveal = Reveal::UserFacing;` modification later on (since `key` is passed immutable).

Reply:
Well the point was that it could just be declared mut in the parameter.

Reply:
Hm, wouldn't that change the semantics? :thinking:

Reply:
It needs to be scoped in this block, because only the copy should be modified, so @matthiaskrgr is right.